sdk/eventhub/event-hubs/src/eventHubBufferedProducerClient.ts

```diff
@@ -0,0 +1,315 @@
+   * <yournamespace>.servicebus.windows.net
+   * @param eventHubName - The name of the specific Event Hub to connect the client to.
+   * @param credential - An credential object used by the client to get the token to authenticate the connection
+   * with the Azure Event Hubs service.
+   * See &commat;azure/identity for creating credentials that support AAD auth.
+   * Use the `AzureNamedKeyCredential` from &commat;azure/core-auth if you want to pass in a `SharedAccessKeyName`
+   * and `SharedAccessKey` without using a connection string. These fields map to the `name` and `key` field respectively
+   * in `AzureNamedKeyCredential`.
+   * Use the `AzureSASCredential` from &commat;azure/core-auth if you want to pass in a `SharedAccessSignature`
+   * without using a connection string. This field maps to `signature` in `AzureSASCredential`.
+   * @param options - A set of options to apply when configuring the client.
+   * - `retryOptions`   : Configures the retry policy for all the operations on the client.
+   * For example, `{ "maxRetries": 4 }` or `{ "maxRetries": 4, "retryDelayInMs": 30000 }`.
+   * - `webSocketOptions`: Configures the channelling of the AMQP connection over Web Sockets.
+   * - `userAgent`      : A string to append to the built in user agent string that is passed to the service.
+   */
+  constructor(
+    fullyQualifiedNamespace: string,
+    eventHubName: string,
+    credential: TokenCredential | NamedKeyCredential | SASCredential,
+    options: EventHubBufferedProducerClientOptions // eslint-disable-line @azure/azure-sdk/ts-naming-options
+  );
+  constructor(
+    fullyQualifiedNamespaceOrConnectionString1: string,
+    eventHubNameOrOptions2: string | EventHubBufferedProducerClientOptions,
+    credentialOrOptions3?:
+      | TokenCredential
+      | NamedKeyCredential
+      | SASCredential
+      | EventHubBufferedProducerClientOptions,
+    options4?: EventHubBufferedProducerClientOptions // eslint-disable-line @azure/azure-sdk/ts-naming-options
+  ) {
+    this._context = createConnectionContext(
+      fullyQualifiedNamespaceOrConnectionString1,
+      eventHubNameOrOptions2,
+      credentialOrOptions3,
+      options4
+    );
+    if (typeof eventHubNameOrOptions2 !== "string") {
+      this._clientOptions = eventHubNameOrOptions2 || {};
+    } else if (!isCredential(credentialOrOptions3)) {
+      this._clientOptions = credentialOrOptions3 || {};
+    } else {
+      this._clientOptions = options4 || {};
+    }
+  }
+
+  /**
+   * Closes the AMQP connection to the Event Hub instance,
+   * returning a promise that will be resolved when disconnection is completed.
+   *
+   * @param options - The set of options to apply to the operation call.
+   * @returns Promise<void>
+   * @throws Error if the underlying connection encounters an error while closing.
+   */
+  async close(options: BufferedCloseOptions = {}): Promise<void> {
+    if (options.flush) {
+      await this.flush(options);
```

Comment:
So the idea would be that this.flush() is safe to call multiple times? So if 'close' throws an error I should feel free to call close() again?

Reply:
Yes;  flush should be safe to call multiple times.